src/Nethermind/Nethermind.Network/P2P/P2PProtocolHandler.cs

```diff
@@ -224,19 +224,24 @@ public async Task<bool> SendPing()
             Stopwatch stopwatch = Stopwatch.StartNew();
 
             CancellationTokenSource delayCancellation = new CancellationTokenSource();
-            Task firstTask = await Task.WhenAny(pongTask, Task.Delay(Timeouts.P2PPing, delayCancellation.Token));
-            _pongCompletionSource = null;
-            if (firstTask != pongTask)
+            try
             {
-                _nodeStatsManager.ReportTransferSpeedEvent(Session.Node, TransferSpeedType.Latency,0);
-                return false;
-            }
+                Task firstTask = await Task.WhenAny(pongTask, Task.Delay(Timeouts.P2PPing, delayCancellation.Token));
+                if (firstTask != pongTask)
```

Comment:
Wont this always be true? WhenAny will return something different than one of its parameters(?)

Reply:
this is a check on whether Delay or pongTask happened first